packages/e2e-test-utils/src/change-site-timezone.js

```diff
@@ -21,7 +21,10 @@ export async function changeSiteTimezone( timezone ) {
 		( element ) => element.options[ element.selectedIndex ].text
 	);
 	await page.select( '#timezone_string', timezone );
-	await page.click( '#submit' );
+	await Promise.all( [
+		page.click( '#submit' ),
+		page.waitForNavigation( { waitUntil: 'networkidle0' } ),
+	] );
```

Comment:
This is to fix an error similar to #27358 becoming more consistent after upgrading puppeteer.

The solution is to use [`waitForNavigation`](https://github.com/puppeteer/puppeteer/blob/v5.5.0/docs/api.md#pagewaitfornavigationoptions) for places that will trigger a page navigation.